src/issues.ts

```diff
@@ -0,0 +1,459 @@
+   */
+  onIssueLabeled(repo: types.Repository, issue: types.Issue, label: string) {
+    // Basic info
+    const org = repo.owner.login;
+    const name = repo.name;
+
+    // Render the issue body
+    const body_html = marked(issue.body);
+
+    // Send a new issue email
+    return this.sendIssueUpdateEmail(repo, issue, {
+      header: `New Issue in label ${label}`,
+      body: body_html,
+      label: label
+    });
+  }
+
+  /**
+   * Send an email when a new comment is added to an issue.
+   */
+  onCommentCreated(repo: types.Repository, issue: types.Issue, comment: types.Comment) {
+    // Trick for testing
+    if (comment.body == "eval") {
+      console.log("HANDLING SPECIAL COMMENT: eval");
+      return this.onNewIssue(repo, issue);
+    }
+
+    const comment_html = marked(comment.body);
+    const body = `
+      <div>
+        <p>@${comment.user.login}:</p>
+        ${comment_html}
+      </div>`;
+
+    return this.sendIssueUpdateEmail(repo, issue, {
+      header: "New Comment",
+      body: body
+    });
+  }
+
+  /**
+   * Send an email when an issue has been updated.
+   */
+  sendIssueUpdateEmail(repo: types.Repository, issue: types.Issue, opts: SendIssueUpdateEmailOpts) {
+    // Get basic issue information
+    const org = repo.owner.login;
+    const name = repo.name;
+    const number = issue.number;
+
+    // See if this issue belongs to any team.
+    const label = opts.label || this.getRelevantLabel(org, name, issue);
+    if (!label) {
+      console.log("Not a relevant label, no email needed.");
+      return Promise.resolve();
+    }
+
+    // Get label email from mapping
+    let recipient;
+    const label_config = this.config.getRepoLabelConfig(org, name, label);
+    if (label_config) {
+      recipient = label_config.email;
+    }
+
+    if (!recipient) {
+      console.log("Nobody to notify, no email needed.");
+      return Promise.resolve();
+    }
+
+    // Get email subject
+    const subject = this.getIssueEmailSubject(issue.title, org, name, label);
+
+    // Send email update
+    return this.email_client.sendStyledEmail(
+      recipient,
+      subject,
+      opts.header,
+      opts.body,
+      issue.html_url,
+      "Open Issue"
+    );
+  }
+
+  /**
+   * Pick the first label from an issue that has a related configuration.
+   */
+  getRelevantLabel(org: string, name: string, issue: types.Issue) {
+    // Make sure we at least have configuration for this repository
+    const repo_mapping = this.config.getRepoConfig(org, name);
+    if (!repo_mapping) {
+      console.log(`No config for ${org}/${name} in: `, this.config);
+      return undefined;
+    }
+
+    // Get the labeling rules for this repo
+    console.log("Found config: ", repo_mapping);
+
+    // Iterate through issue labels, see if one of the existing ones works
+    // TODO(samstern): Deal with needs_triage separately
+    // TODO(abehaskins): Get clarification here
```

Comment:
This block basically loops through all labels in the repo config and then checks to see if any of them has already been applied to this issue.  If so, we use that one rather than trying to parse the issue and determine the label.